includes/admin/class-wc-rest-payments-files-controller.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Class WC_REST_Payments_Files_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for files.
+ */
+class WC_REST_Payments_Files_Controller extends WC_Payments_REST_Controller {
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/file';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'upload_file' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+			]
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<file_id>\w+)',
+			[
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'get_file' ],
+				'permission_callback' => [],
+			]
+		);
+	}
+
+	/**
+	 * Create file and respond with file object via API.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function upload_file( $request ) {
+		return $this->forward_request( 'upload_evidence', [ $request ] );
+	}
+
+	/**
+	 * Retrieve a file content via API.
+	 *
+	 * @param WP_REST_Request $request - request object.
+	 *
+	 * @return WP_Error|WP_HTTP_Response
+	 */
+	public function get_file( WP_REST_Request $request ) {
+		$file_id = $request->get_param( 'file_id' );
+		$result  = $this->forward_request( 'get_file_contents', [ $file_id ] );
+
+		if ( $result instanceof WP_Error ) {
```

Comment:
[is_wp_error](https://developer.wordpress.org/reference/functions/is_wp_error/) could be helpful here.